### PR TITLE
Handle for and while loops

### PR DESCRIPTION
**Pull request by GuillaumeGen**
Add "for", "to" and "while" keywords, as well as indentation related to "do" and "done".

### CODE REVIEW

File: tests/samples/input/ocaml.ml [hunk 1]
```diff
@@ -495,3 +495,17 @@ let add_as_fun_multiline = fun x ->
   x
 
 let add_as_fun_one_line = fun x -> x
+
+let sum_of_int n =
```
Comment (torhovland) [inline tests/samples/input/ocaml.ml:499]: @aspiwack I'll approve this, but some multi-line elements are turned into single-line here, so you might want to take a look too.

**Review by torhovland — APPROVED** [review]

File: tests/samples/input/ocaml.ml (hunk 1 above)
Comment (aspiwack) [reply tests/samples/input/ocaml.ml:499]: Indeed, there is definitely something fishy here. Even with the most generous interpretation of what a node is, this should not become a single line. Can someone open an issue to track that let-ins seem to attract things upward like this?

(in this particular case, I expect `sum_of_int` to be formatted as written)

Comment (GuillaumeGen) [reply tests/samples/input/ocaml.ml:499]: I can open an issue about it.

Comment (aspiwack) [reply tests/samples/input/ocaml.ml:499]: :arrow_right: #134 .

Comment (torhovland) [reply tests/samples/input/ocaml.ml:499]: This can be merged, then.

**Pull request merged**